NumLib/TimeStepping/Algorithms/FixedTimeStepping.cpp

```diff
@@ -31,16 +31,54 @@ FixedTimeStepping::FixedTimeStepping(double t0, double tn, double dt)
 : _t_initial(t0), _t_end(tn), _dt_vector(static_cast<std::size_t>(std::ceil((tn-t0)/dt)), dt), _ts_prev(t0), _ts_current(t0)
 {}
 
-FixedTimeStepping*
+std::unique_ptr<ITimeStepAlgorithm>
 FixedTimeStepping::newInstance(BaseLib::ConfigTree const& config)
 {
     config.checkConfParam("type", "FixedTimeStepping");
 
     auto const t_initial = config.getConfParam<double>("t_initial");
     auto const t_end     = config.getConfParam<double>("t_end");
-    auto const dt        = config.getConfParam<double>("dt");
-
-    return new FixedTimeStepping(t_initial, t_end, dt);
+    auto const delta_ts  = config.getConfSubtree("timesteps");
+
+    std::vector<double> timesteps;
+    double t_curr = t_initial;
+    double delta_t = 0.0;
+
+    // TODO: consider adding call "listNonEmpty" to config tree
+    auto const range = delta_ts.getConfSubtreeList("pair");
+    if (range.begin() == range.end()) {
+        ERR("no timesteps have been given");
+        std::abort();
+    }
+    for (auto const pair : range)
+    {
+        auto const repeat = pair.getConfParam<std::size_t>("repeat");
+        delta_t           = pair.getConfParam<double>("delta_t");
+
+        if (repeat == 0) {
+            ERR("<repeat> is zero.");
+            std::abort();
+        }
+        if (delta_t < 0.0) {
+            ERR("timestep <delta_t> is smaller than zero.");
+            std::abort();
+        }
+
+        if (t_curr <= t_end) {
```

Comment:
I'd leave it as `<=`, because here we use `t_curr += repeat * delta_t`, but the time loop later on will use `t_curr += delta_t` repeatedly. Due to rounding errors those two methods might get to different results. Maybe one should even add a small epsilon in order to make sure that the generated timestep list always covers the entire simulation time.